Add an info summary on standard output.

iwonbigbro opened the pull request:
This is to allow build runners like those on GitLab and GitHub to parse
the build log for the coverage score and display it on the build results
page.

Commit:
Add an info summary on standard output.

This is to allow build runners like those on GitLab and GitHub to parse
the build log for the coverage score and display it on the build results
page.

SimonKagstrom commented:
Thanks!

However, I'd prefer to add something like this for kcov instead. I've deprecated shcov in favor of kcov, which supports much more features (and Python, compiled program coverage collection) than shcov did.

For kcov, I suppose it could be done as an external Python script that parses the cobertura XML output (I use this for the internal testing already, albeit not for summaries). Do you want to take a shot at that?

Commit:
Make shcov more POSIX compliant.

Use the TMPDIR environment setting for the default location of /tmp and
also provide an override environment setting for the output directory.

SimonKagstrom commented:
I've merged your changes, thanks!

But kcov is really the way to go in the future. It's also available on github, here

   https://github.com/SimonKagstrom/kcov

iwonbigbro commented:
I had looked at kcov, but i am running shcov in an unalterable build environment. I didn't want the overhead of the c compilation for this particular project and may not have the appropriate dependencies to do so. For portability, shcov works well enough, despite not having branch coverage analysis.

I will definitely look to use kcov for future less mature projects.

SimonKagstrom commented:
I understand, and thanks for the patches! Good to hear that someone is actually using shcov as well :-)

Anyway, if you need an example for the future, this project, https://github.com/vlajos/misspell_fixer, is using kcov for bash scripts together with travis-ci.org (if you're using that for continuous integration).

iwonbigbro commented:
Cool, thanks.
